docs/user_guide/ml.md

```diff
@@ -1,8 +1,254 @@
+At the end we transform the awkward array into a Tensorflow tensor, something that our machine learning model can handle.
+```{literalinclude} ./ml_code.py
+:language: python
+:start-at: def prepare_events
+:end-at: return tf.convert_to_tensor
+```
+The actual building of the model is also handled by a separate function.
+```{literalinclude} ./ml_code.py
+:language: python
+:start-at: def build_model
+:end-at: return model
+```
+With this little example one can see why it is of good practice to separate this process into small chunks.
+
+## evaluate:
+Within {py:meth}`~columnflow.ml.MLModel.train` one defined the trainings process of the ML model, while in {py:meth}`~columnflow.ml.MLModel.evaluate` its evaluation is defined.
+The corresponding `task` is {py:class}`~columnflow.tasks.ml.MLEvaluation` and depends on {py:class}`~columnflow.tasks.ml.MLTraining` and will therefore trigger a training if no training was performed before.
+
+For each fold of the k-folds a neural network model is trained and can be accessed from `models`.
+The actual loading, of the trained model stored in the list, is defined in {py:meth}`~columnflow.ml.MLModel.open_model` function.
+
+The awkward array `events` is loaded in chunks and contains the merged events of all folds.
+To filter out the test set, create a mask with `fold_indices`.
+
+If you want to preserve columns and write them out into a parquet file, append the columns to `events` using `set_ak_column` and return `events`.
+All columns not present in {py:meth}`~columnflow.ml.MLModel.produces` are then filtered out.
+
+In the following example, the models prediction as well as the number of muons and electrons are saved, all the other columns in events are thrown away, since they are not present in `use`:
+
+Don't confuse this behavior with the parameter `events_used_in_training`.
+This flag determines if a certain `dataset` and shift combination can be used by the task.
+
+```{literalinclude} ./ml_code.py
+:language: python
+:start-at: def evaluate
+:end-at: return events
+```
+
+The evaluations output is saved as parquet file with the name of the model as field name.
+To get the files path afterwards, rerun the same law command again, but with `--print-output 0` at the end.
+
+# Commands to start training and evaluation
+To start the machine learning training `law run cf.MLTraining`, while for evaluation use `law run cf.MLEvaluation`.
+When not using the `config`, `calibrators`, `selector`s or `dataset`
+```bash
+law run cf.MLTraining \
+    --version your_current_version \
+    --ml-model test_model_name \
+    --config run2_2017_nano_uhh_v11_limited \
+    --calibrators calibrator_name \
+    --selector selector_name \
+    --dataset datasets_1,dataset_2,dataset_3,...
+
+law run cf.MLEvaluation \
+    --version v1 \
+    --ml-model test_model \
+    --config run2_2017_nano_uhh_v11_limited \
+    --calibrators skip_jecunc
+```
```

Comment:
I would split this into two boxes (makes copying easier)

Reply:
Can you split this into two boxes?

Reply:
already done. This didnt showed up here because i marked the code block by myself, which counted as NEW comment. On this new comment the suggestion was applied to.